### PR TITLE
Adds create short link functionality to FDL quickstart

### PR DESCRIPTION
**Pull request by cavalos0086**
Updates the FDL Quickstart to generate a short dynamic link to have the same behavior as the iOS counterpart.

### CODE REVIEW

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java [hunk 1]
```diff
@@ -165,4 +181,39 @@ private void validateAppCode() {
...
+            public void onComplete(@NonNull Task<ShortDynamicLink> task) {
+              if (task.isSuccessful()) {
+                // Short link created
+                Uri shortDynamicLink = task.getResult().getShortLink();
+                TextView shortLinkTextView = findViewById(R.id.shortLinkViewSend);
+                shortLinkTextView.setText(shortDynamicLink.toString());
+              } else {
+                // Error
+                Log.e(TAG, task.getException().getMessage());
+                throw new Error(task.getException().getMessage());
```
Comment (thatfiredev) [inline dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java:214]: Simply logging the error should be fine, there's no need to re-throw it.
That's the approach we use across all quickstarts.

```suggestion
                Log.e(TAG, task.getException().getMessage());
```

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt [hunk 2]
```diff
@@ -106,6 +118,29 @@ class MainActivity : AppCompatActivity() {
...
+        androidParameters {
+            minimumVersion = minVersion
+        }
+        }.addOnSuccessListener { (shortLink, flowchartLink) ->
+        val shortLinkTextView = findViewById<TextView>(R.id.shortLinkViewSend);
+        shortLinkTextView.text = shortLink.toString();
+        }.addOnFailureListener(this) { e ->
+        Log.e(TAG, e.toString());
+        throw java.lang.Error(e.toString());
+        }
```
Comment (thatfiredev) [inline dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt:141]: The indentation is looking odd here, I wonder how `ktlint` didn't catch that (maybe we didn't configure it to warn about indentation improvements).
We also don't need to throw the exception:
```suggestion
        Firebase.dynamicLinks.shortLinkAsync {
            link = deepLink
            domainUriPrefix = uriPrefix
            androidParameters {
                minimumVersion = minVersion
            }
        }.addOnSuccessListener { (shortLink, flowchartLink) ->
            val shortLinkTextView = findViewById<TextView>(R.id.shortLinkViewSend);
            shortLinkTextView.text = shortLink.toString();
        }.addOnFailureListener(this) { e ->
            Log.e(TAG, e.toString());
        }
```

**Review by thatfiredev — CHANGES_REQUESTED** [review]

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java (hunk 1 above)
Comment (cavalos0086) [reply dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java]: Thanks for the comment. I removed the throw statements

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt (hunk 2 above)
Comment (cavalos0086) [reply dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt]: Yea that is odd. I wonder if we could add an editorConfig to remove this issues with indentation. Any thoughts?

Comment (thatfiredev) [reply dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt]: <kbd>cmd</kbd> + <kbd>opt</kbd> + <kbd>L</kbd> or <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>L</kbd> usually does the trick for me on Android Studio.
But if you know the editConfig to enforce it, feel free to send a PR and I'll review it :)

**Review by thatfiredev — APPROVED** [review]
LGTM.
Thanks for making sure the android and iOS quickstarts are on pair ,@cavalos0086 ! :)

**Review by eldhosembabu — CHANGES_REQUESTED** [review]
Please address the comments. Thanks

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java [hunk 3]
```diff
@@ -59,13 +63,25 @@ protected void onCreate(Bundle savedInstanceState) {
         final Uri deepLink = buildDeepLink(Uri.parse(DEEP_LINK_URL), 0);
         linkSendTextView.setText(deepLink.toString());
 
+        // create a short link and display it in the UI
+        buildShortLinkFromParams(Uri.parse(DEEP_LINK_URL), 0);
```
[the remark this answers is not in the document]
Comment (cavalos0086) [reply dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java]: As we spoke offline, I will change this to button onClick instead. The reason I did it as a page refresh is to have parity with the long link call. But, as you pointed out, shortLink is a write request and it should be a button click.

File: dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt [hunk 4]
```diff
@@ -35,10 +38,19 @@ class MainActivity : AppCompatActivity() {
         val newDeepLink = buildDeepLink(Uri.parse(DEEP_LINK_URL), 0)
         linkSendTextView.text = newDeepLink.toString()
 
+        // create a short link and display it in the UI
+        buildShortLinkFromParams(Uri.parse(DEEP_LINK_URL), 0)
```
[the remark this answers is not in the document]
Comment (cavalos0086) [reply dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainActivity.kt]: Fixed.

**Review by eldhosembabu — APPROVED** [review]

**Pull request merged**